### DISCUSSION TITLE
Is this library meant as a replacement for Rg.Plugins.Popup?

### DISCUSSION THREAD

**Discussion by mr5z** [LuckyDucko/Mopups#101] [General]
The only reason I ended up here is due to the linked references from Rg.Plugins.Popup, & Prism Popups pointing to this repository.
It seems to be a replacement until I tried to migrate my library & app using this.

The examples on readme tells me that it's trying to be more than a popup and an extension to Prism by auto generating ViewModel instances which seems to violate Single Responsibility principle.

I thought I could use this as a replacement for the aforementioned libraries, if not, please enlighten. Thanks!

**Comment by IeuanWalker** [comment]
@mr5z Hi, im not sure why the readme is only showing the viewmodel approach, but you can still use this library like the XF version

I.e.
`PopupNavigation.Instance.PushAsync(new YourPopupPage());`

**Reply by mr5z** [reply]
@IeuanWalker hey thanks for the reply. I have another question and it's okay if you don't have an answer yet.
Do you think I can publish a public repo and public NuGet package using this library?
It'll be just an extension (dependency of this package) and I'm not going to use it for any monetary means fwiw.

**Reply by IeuanWalker** [reply]
@mr5z, its best to ask @LuckyDucko he's the owner/ maintainer of this project.

From my POV if your package is just extending/ adding new features and is just referencing this nugget package. I don't see an issue with it.

But if you are cloning this project/ code and repackaging it, then I'd rethink it if I were you. But I don't believe anything is stopping you as it's quite an open license - https://github.com/LuckyDucko/Mopups/blob/master/LICENSE

If you did want more features in this library though, consider opening an issue detailing what you want. From my experience, @LuckyDucko is pretty open to PRs, but worth starting an issue to discuss first.

**Reply by mr5z** [reply]
I don't think a PR would be a valid since it will reference the Prism library so I think a separate library would make more sense.
I have created private repo atm. Will wait for @LuckyDucko response to determine which license is best suited (I'm planning GNU GPL v3 if possible) before making it public.

**Comment by LuckyDucko** [comment]
Yeah this library is the replacement for RG. I dont actively develop for mobile anymore, so i dont do as much dev as i like on it (still need to do the latest release! My bad)

You can completely ignore any of the viewmodel things I've added. Thats just some special sauce added on to make things super easy peasy for people who just want to add this library and fire out a popup premade for them. No mess no fess.

Except it does mean people who have no interest in the special sauce have this code bogging down the dll's. 

Let me know your thoughts on any changes that'll help you out.
